### PR TITLE
Fix `QuantileForecast.quantile` in case only mean is stored

### LINKED ISSUE #2498: AssertionError while using ets from wrapper.
## Description
ets from statforecast model producing error.

## To Reproduce
(Please provide minimal example of code snippet that reproduces the error. For existing examples, please provide link.)

```python
from gluonts.ext.statsforecast import ETSPredictor
from gluonts.dataset.repository.datasets import get_dataset
from gluonts.evaluation.backtest import backtest_metrics
dataset = get_dataset("pedestrian_counts")
predictor = ETSPredictor(prediction_length=dataset.metadata.prediction_length)
agg_metrics, agg_perite = backtest_metrics(dataset.test, predictor)

```

## Error message or code output
(Paste the complete error message, including stack trace, or the undesired output that the above snippet produces.)

```
RemoteTraceback                           Traceback (most recent call last)
RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/usr/lib/python3.7/multiprocessing/pool.py", line 121, in worker
    result = (True, func(*args, **kwds))
  File "/usr/lib/python3.7/multiprocessing/pool.py", line 44, in mapstar
    return list(map(*args))
  File "/home/bhaskar/Desktop/gluonts/src/gluonts/evaluation/_base.py", line 56, in worker_function
    return evaluator.get_metrics_per_ts(ts, forecast)
  File "/home/bhaskar/Desktop/gluonts/src/gluonts/evaluation/_base.py", line 343, in get_metrics_per_ts
    median_fcst = forecast.quantile(0.5)
  File "/home/bhaskar/Desktop/gluonts/src/gluonts/model/forecast.py", line 726, in quantile
    quantiles, quantile_predictions
  File "/home/bhaskar/Desktop/gluonts/src/gluonts/model/forecast.py", line 52, in __init__
    ), "Need at least two points for linear interpolation."
AssertionError: Need at least two points for linear interpolation.
"""
```

## Environment
- Operating system:Ubuntu 18.04
- Python version:3.7
- GluonTS version:dev
- MXNet version:

(Add as much information about your environment as possible, e.g. dependencies versions.)

### PR DESCRIPTION
**Pull request by lostella**
*Issue #, if available:* Fixes #2498

*Description of changes:* There was a bug into the logic that decided whether to do linear interpolation or not, in the edge case where only mean is stored in the forecast object. This fixes it and adds a test.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Please tag this pr with at least one of these labels to make our release process faster:** BREAKING, new feature, bug fix, other change, dev setup

### CODE REVIEW

File: test/model/test_forecast.py [hunk 1]
```diff
@@ -75,6 +75,23 @@ def percentile(value):
...
+        forecast_arrays=np.ones(shape=(1, 12)),
+        start_date=pd.Period("2022-03-04 00", freq="H"),
+        forecast_keys=["mean"],
+    )
+
+    assert forecast.prediction_length == 12
+    assert len(forecast.index) == 12
+    assert forecast.index[0] == pd.Period("2022-03-04 00", freq="H")
+
+    for level in [0.1, 0.5, 0.7]:
```
Comment (melopeo) [inline test/model/test_forecast.py:89]: Ok. It seems that the main message here is that, if we only have the field `mean` in the `QuantileForecast` then we will return `np.nan` if any quantile is requested. LGTM.

**Review by melopeo — APPROVED** [review]

File: test/model/test_forecast.py (hunk 1 above)
Comment (lostella) [reply test/model/test_forecast.py:89]: Exactly

**Comment by jaheba** [comment]
What looks weird to me is that we treat `mean` as a quantile. Shouldn't we just keep them separate?

**Comment by lostella** [comment]
> What looks weird to me is that we treat `mean` as a quantile. Shouldn't we just keep them separate?

Yes, I agree. Either that or `QuantileForecast` gets a different name.
